samples/cpp/text_generation/README.md

```diff
@@ -54,6 +54,17 @@ The following template can be used as a default, but it may not work properly wi
 "chat_template": "{% for message in messages %}{% if (message['role'] == 'user') %}{{'<|im_start|>user\n' + message['content'] + '<|im_end|>\n<|im_start|>assistant\n'}}{% elif (message['role'] == 'assistant') %}{{message['content'] + '<|im_end|>\n'}}{% endif %}{% endfor %}",
 ```
 
+#### NPU support
+
+NPU device is supported with some limitations. See [NPU inference of
+LLMs](https://docs.openvino.ai/2024/learn-openvino/llm_inference_guide/genai-guide-npu.html) documentation. In particular:
+
+- Models must be exported with symmetric INT4 quantization (`optimum-cli export openvino --weight-format int4 --sym --model <model> <output_folder>`).
+  For models with more than 4B parameters, channel wise quantization should be used (`--group-size -1`).
+- Only greedy search is supported (`do_sample` must be set to False)
```

Comment:
it's not a case anymore.

NPU uses generic Sampler functionality and supports do_sample as well.

The only limitation is:
- beam search is not supported as requires dynamic batch support
- parallel sampling is not supported for the same reason.

Reply:
Thanks, that is good news. Updated the README and added to use at least 2025.0. Until recently Llama models were broken with nightly so NPU users had to use 2024.6 - but the issue was fixed in 2025.0 RC2.